src/shape.rs

```diff
@@ -348,3 +376,356 @@ mod test
         assert_eq!(None, r1.intersect(&r2));
     }
 }
+
+///////////////////////////////////
+
+/// A struct representing an axis-aligned rounded rectangle. Two points and a float are
+/// stored: the top left vertex, the bottom right vertex and the radius of the
+/// rounded corners.
+///
+/// Alias for a rectangle with u32 coordinates.
+pub type URoundRect = RoundedRectangle<u32>;
+
+/// A struct representing an axis-aligned rounded rectangle. Two points and a float are
+/// stored: the top left vertex, the bottom right vertex and the radius of the
+/// rounded corners.
+///
+/// Alias for a rectangle with i32 coordinates.
+pub type IRoundRect = RoundedRectangle<i32>;
+
+/// A struct representing an axis-aligned rounded rectangle. Two points and a float are
+/// stored: the top left vertex, the bottom right vertex and the radius of the
+/// rounded corners.
+///
+/// Alias for a rectangle with f32 coordinates.
+pub type RoundRect = RoundedRectangle<f32>;
+
+/// A struct representing an axis-aligned rounded rectangle. Two points and a float are
+/// stored: the top left vertex, the bottom right vertex and the radius of the
+/// rounded corners.
+#[derive(Debug, PartialEq, Eq, Clone)]
+#[repr(C)]
+pub struct RoundedRectangle<T = f32>
+{
+    top_left: Vector2<T>,
```

Comment:
Rather than having the fields `top_left` and `bottom_right`, I think it would be good to use a `Rectangle` here. So this struct would contain two fields -- a `Rectangle` and a radius.